### PR TITLE
Cross val, class and doc strings

### PR DESCRIPTION
**Pull request by dantoll279**

### CODE REVIEW

File: forecasting/auto_ts.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        """
+        :param data: pandas.DataFrame - Historical time series data with a date-time index.
+        :param target_col: str - Column with historical data. If none will default to the first column.
+        :param period: int - Seasonal period.
+        """
+
+        self.data = data
+        self.period = period
+
+        if not target_col:
```
Comment (BenH235) [inline forecasting/auto_ts.py]: Maybe remove and insert error handling

File: forecasting/auto_ts.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+            target_col = self.data.columns[0]
+
+        self.target_col = target_col
+
+    def select_best_model(self, n_splits:int=5, test_size:int=None, models:dict=None, time_taken:bool=False):
```
Comment (BenH235) [inline forecasting/auto_ts.py]: Add minimisation error score as argument (string input). Add available strings in docstring.

Always return time_taken inside dataframe (remove bool option)

File: forecasting/model_selection.py [hunk 3]
```diff
@@ -1,29 +1,169 @@
...
+
+
+
+        #creating a list to store each fold
+        cv_summary = []
+
+        forecaster = models[model]
+
+        if time_taken:
+            print(f'{model}:')
```
Comment (BenH235) [inline forecasting/model_selection.py]: remove this

File: forecasting/model_selection.py [hunk 4]
```diff
@@ -1,29 +1,169 @@
...
+                                  **kwargs)
+
+            cv_output['forecast'] = forecast['forecast'].values
+            cv_output['fold'] = fold
+            cv_output['error'] = cv_output[target_col] - cv_output['forecast']
+            cv_summary.append(cv_output)
+            
+            end = time.time()
+
+            if time_taken:
```
Comment (BenH235) [inline forecasting/model_selection.py]: remove

File: forecasting/model_selection.py [hunk 5]
```diff
@@ -1,29 +1,169 @@
...
+                print('fold {fold_no}: {time:.4f}'.format(fold_no=fold,time=end-start))
+
+        output_dict[model] = pd.concat(cv_summary)
+
+
+    output_frame = pd.concat(output_dict,names = ['model',])
```
Comment (BenH235) [inline forecasting/model_selection.py]: Try axis = 0/1 depending on which output dataframe works best

File: forecasting/more_models.py [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+from prophet import Prophet
+from statsforecast.models import AutoETS, AutoARIMA
+from .normal_naive_models import naive_pi, drift_pi, mean_pi
+from .bootstrap_naive_models import forecast_dates, naive_error, drift_error, mean_error
+from sklearn.metrics import *
+from statsmodels.tsa.seasonal import MSTL
+
+
+
+def prophet_fit(df:pd.DataFrame, target_col:str,**kwargs) -> pd.DataFrame:
```
Comment (BenH235) [inline forecasting/more_models.py]: Add Prophet() class. Remove prophet_fit(), prohet_forecast() and have these as seperate methods (.fit(), .predict()). Possibly read through paper to see if easier way to extract prediciton intervals.

**Review by BenH235 — CHANGES_REQUESTED** [review]
